daemon_linux_systemv.go

```diff
@@ -235,23 +235,29 @@ lockfile="/var/lock/subsys/$proc"
 stdoutlog="/var/log/$proc.log"
 stderrlog="/var/log/$proc.err"
 
-[[ -d $(dirname $lockfile) ]] || mkdir -p $(dirname $lockfile)
+[ -d $(dirname $lockfile) ] || mkdir -p $(dirname $lockfile)
 
 [ -e /etc/sysconfig/$proc ] && . /etc/sysconfig/$proc
 
 start() {
     [ -x $exec ] || exit 5
 
+    if [ -f $pidfile ]; then
+        if ! [ -d "/proc/$(cat $pidfile)" ]; then
+            rm $pidfile
+        fi
+    fi
+
```

Comment:
Pid file existence check is smart. But that files don't always contain valid pids. So I have added code that checks if pid is valid(running) or not. And if not running, it deletes the pid file.